package kr.ac.kopo.psjjj._026example.repository;

import jakarta.transaction.Transactional;
import kr.ac.kopo.psjjj._026example.domain.Member3;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.beans.Transient;
import java.util.List;

public interface Member3Repository2 extends JpaRepository<Member3, Integer> {


    // 전체 행(entity)를 조회
    @Transactional
    @Query(value = "select * from Member3", nativeQuery = true)
    // @query(value = "select entity from Member3 entity") // JPQL
    public List<Member3> selectMembers();

    // 특정 id를 가진 한 행 조회
    @Transactional
    @Query(value = "select entity from Member3 entity where id = :e_id", nativeQuery = true)
    public Member3 selectById(@Param("e_id")int id);

    // Member3 삽입
    @Transactional
    @Query(value = "insert into Member3(name, age, email) values(:e_name, :e_age, :e_email)")
    public int insertMember(@Param("e_name") String name,@Param("e_age") int age,@Param("e_email") String email);

    // Member3 수정
    @Transactional
    @Modifying
    @Query(value = "update Member3 set name=?, age=?, email=? where id=?", nativeQuery = true)
//    @Query(value = "update Member3 set name=:e_name, age=:e_age, email=:e_email where id=:e_id")
    public int updateMember(@Param("e_name") String name,@Param("e_age") int age,@Param("e_email") String email, @Param("e_id")int id);

    // Member3 삭제
    @Transactional
    @Modifying
    @Query(value = "delete from Member3 where id=?", nativeQuery = true)
//    @Query(value = "delete from Member3 where id=:e_id")
    public int deleteMember(@Param("e_id")int id);
}
